src/components/shapes/helpers.js

```diff
@@ -92,9 +92,10 @@ exports.getPixelToData = function(gd, axis, isVertical, opt) {
     if(axis) {
         if(opt === 'domain') {
             pixelToData = function(p) {
+                var q = p - axis._offset;
```

Comment:
One could still divide by `axis._length` when computing the temporary variable. Right?
```js
pixelToData = function(p) {
    var q = (p - axis._offset) / axis._length;
    return isVertical ? (1 - q) : q;
};
```

Reply:
d70f4e9c79bb1e8bd5a28002847c60b502da6884 fixes